cpp/benchmarks/string/convert_datetime_benchmark.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <benchmarks/common/generate_benchmark_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/synchronization/synchronization.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/strings/convert/convert_datetime.hpp>
+#include <cudf/wrappers/timestamps.hpp>
+
+class StringDateTime : public cudf::benchmark {
+};
+
+enum class direction { to, from };
+
+template <class TypeParam>
+void BM_convert_datetime(benchmark::State& state, direction dir)
+{
+  cudf::size_type const n_rows{static_cast<cudf::size_type>(state.range(0))};
+  auto const data_type = cudf::data_type(cudf::type_to_id<TypeParam>());
```

Comment:
Redundant `cudf::size_type`
```suggestion
  auto const n_rows    = static_cast<cudf::size_type>(state.range(0));
  auto const data_type = cudf::data_type(cudf::type_to_id<TypeParam>());
```